### PR TITLE
Enable branch coverage

### PR DESCRIPTION
**Pull request by hunchr**
TICKET-21097

### CODE REVIEW

**Review by rnestler — APPROVED** [review]

**Review by coorasse — CHANGES_REQUESTED** [review]
I think we should set a minimum coverage to 100, as per line coverage. Teams can always decide to reduce it in their project or to exclude branches. Enabling without enforcing brings no value. It will simply be ignored.

**Comment by hunchr** [comment]
> I think we should set a minimum coverage to 100, as per line coverage. Teams can always decide to reduce it in their project or to exclude branches. Enabling without enforcing brings no value. It will simply be ignored.

the coverage was already set to 100% before
```rb
SimpleCov.minimum_coverage 100
```

**Review by coorasse — COMMENTED** [review]
Can you please explain to me what `primary_coverage` does? I expect a `SimpleCov.minimum_branch_coverage` option since they are two separate stats.

File: ruby_on_rails/rspec.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ You should know exactly why you are adding each one of them and why is necessary
     add_filter 'app/mailers/application_mailer.rb'
     add_filter 'app/models/application_record.rb'
     add_filter '.semaphore-cache'
+    enable_coverage :branch
+    primary_coverage :branch
   end
   SimpleCov.minimum_coverage 100
```
Comment (rnestler) [inline ruby_on_rails/rspec.md:44]:
```suggestion
  SimpleCov.minimum_coverage line: 100, branch: 100
```

This is how you configure branch coverage.

**Review by rnestler — CHANGES_REQUESTED** [review]

File: ruby_on_rails/rspec.md (hunk 1 above)
Comment (hunchr) [reply ruby_on_rails/rspec.md]: Interesting, this is never mentioned in the simplecov docs

**Comment by hunchr** [comment]
> Can you please explain to me what `primary_coverage` does? I expect a `SimpleCov.minimum_branch_coverage` option since they are two separate stats.

I disabled `primary_coverage` again and used @rnestler's suggestion

File: ruby_on_rails/rspec.md (hunk 1 above)
Comment (coorasse) [reply ruby_on_rails/rspec.md]: https://github.com/simplecov-ruby/simplecov#minimum-coverage

**Review by coorasse — APPROVED** [review]
This looks good now!

**Pull request merged**